### PR TITLE
Constants in jhipster tests

### LINKED ISSUE #14235: Ease generator understanding
##### **Overview of the feature request**

Not really a request but more a plan for what I'm trying to currently achieve.

The plan is divided into multiple steps and I think we gonna win some things at each of them.

1. Use constants in all js files.
As you may have noticed, I'm currently introducing options constants anywhere I can in the generators js files, replacing `buildTool == 'maven'` by `buildTool == MAVEN` and this for all constant.
As a developer with an IDE, I love the folding capability to see where a particular constant is used.
Using constants instead of plain string will help to identify all the generators, needles and helpers that uses a jhipster option.
Also, I think it will help the test maintenance and the removal of some jdl historic syntax mismatch (for example the `false`vs the `no`).

2. Align the generator's context feeding

Context is populated by options differently on each generator: 
for example in the [_loading()](https://github.com/jhipster/generator-jhipster/blob/e22ca7cfd00f2a905ae7996d7adfcf82096bb4e5/generators/client/index.js#L163) method in the client generator whereas its in the [getConfig()](https://github.com/jhipster/generator-jhipster/blob/e22ca7cfd00f2a905ae7996d7adfcf82096bb4e5/generators/ci-cd/index.js#L90) one in the ci-cd generator.

**Question**: Can someone tell me which is the generator that uses the best practices so that I'll replicate it in all of them? I'm not a Yeoman black belt, and I'm afraid to do it wrong.

So that we'll have something that is replicable and begin to mutualize some methods names and calls. Also it will be easier for everyone to have a standard canvas/template for method override and extension (I'm thinking of blueprints).

3. Use 'derived properties' in templates instead of using logic (equality tests, string tests)

The final step would be to use only boolean logic in templates, thus injecting derived attributes in generators context:
for example, injecting a `ìsBuildToolMaven` instead of using `buildTool == 'maven'`in all the ejs.
This last step will bring additional value on top of the two previous one (and can only be achieved well with the two previous one implemented):

- Easy folding: a search in the IDE will lead to all check achieved on an option
- Templates will be a little bit cleaner and just be dumb as possible (only containing OR and AND ejs checks)
- Also, any newcomer will be able to see where using the 'maven' 'buildTool' option is impacting the generation, so it will be easier for him to either add a similar option (sbt, bnd, ant :-p) or to override some option's behaviour.
- Finally, it should be way easier for the core committer to maintain an option, and may be to extract them in their dedicated templates (I would love to have template composition depending on options one day).

##### **Motivation for or Use Case**

Well known and adopted frameworks are very clean at their core (Spring, Hibernate) so that they can scale and adapt more easily.

##### **Related issues or PR**

[14206](https://github.com/jhipster/generator-jhipster/pull/14206)
[14203](https://github.com/jhipster/generator-jhipster/pull/14203)

Last request: can you please be reactive on MR contributing to this request please?
This is a huge work and impacting all the repo's file: there's a high risk each time to face many merge conflict and also no one would want to review all the files of the generator at once: it would be better to accept it by small increment...

If someone else want to contribute to the initiative, please come! There's easily a room for many of us (also, if new PR could anticipate, \o/).

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by Tcharl**
closes #14235 : You know what? I'm haaaaaappy :-)

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: test/cypress.spec.js [hunk 1]
```diff
@@ -36,7 +34,7 @@ describe('jhipster:cypress', () => {
   });
 
   describe('generate cypress with React client with OAuth2', () => {
-    const options = { testFrameworks: ['cypress'], clientFramework: REACT, authenticationType: 'oauth2' };
+    const options = { testFrameworks: ['cypress'], clientFramework: REACT, authenticationType: OAUTH2 };
```
Comment (DanielFran) [inline test/cypress.spec.js:37]:
```suggestion
    const options = { testFrameworks: [CYPRESS], clientFramework: REACT, authenticationType: OAUTH2 };
```

File: test/cypress.spec.js [hunk 2]
```diff
@@ -60,7 +58,7 @@ describe('jhipster:cypress', () => {
   });
 
   describe('generate cypress with Angular client with JWT', () => {
-    const options = { testFrameworks: ['cypress'], clientFramework: ANGULAR, authenticationType: 'jwt' };
+    const options = { testFrameworks: ['cypress'], clientFramework: ANGULAR_X, authenticationType: JWT };
```
Comment (DanielFran) [inline test/cypress.spec.js:61]:
```suggestion
    const options = { testFrameworks: [CYPRESS], clientFramework: ANGULAR_X, authenticationType: JWT };
```

File: test/cypress.spec.js [hunk 3]
```diff
@@ -85,7 +83,7 @@ describe('jhipster:cypress', () => {
   });
 
   describe('generate cypress with Angular client with OAuth2', () => {
-    const options = { testFrameworks: ['cypress'], clientFramework: ANGULAR, authenticationType: 'oauth2' };
+    const options = { testFrameworks: ['cypress'], clientFramework: ANGULAR_X, authenticationType: OAUTH2 };
```
Comment (DanielFran) [inline test/cypress.spec.js:86]:
```suggestion
    const options = { testFrameworks: [CYPRESS], clientFramework: ANGULAR_X, authenticationType: OAUTH2 };
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Comment by Tcharl** [comment]
Thank you Daniel!

**Comment by Tcharl** [comment]
Are you okl with this one @DanielFran ?

**Review by DanielFran — APPROVED** [review]

**Pull request merged**